src/block/Liquid.php

```diff
@@ -366,7 +366,7 @@ protected function checkForHarden() : bool{
 	}
 
 	protected function liquidCollide(Block $cause, Block $result) : bool{
-		$ev = new BlockFormEvent($this, $result);
+		$ev = new BlockFormEvent($this, $result, $cause);
```

Comment:
You might want to test and verify this actually does what you think. This parameter was never previously used, so if it wasn't passed correctly, no one will know.

Reply:
This method is only called within the [`Lava` block class](https://github.com/pmmp/PocketMine-MP/blob/next-major/src/block/Lava.php) in those four cases:
(I verified all four of them by using some `var_dump`s.)
- [`Lava` -> `Obsidian`](https://github.com/pmmp/PocketMine-MP/blob/next-major/src/block/Lava.php#L76)
A lava source collides with any water block and forms obsidian.
```
Target Block: 		string(4) "Lava"
New State:		string(6) "Obsidian"
Causing Block: 		string(8) "Water"
```
- [`Lava` -> `Cobblestone`](https://github.com/pmmp/PocketMine-MP/blob/next-major/src/block/Lava.php#L79)
A non-source lava block collides with any water block and forms cobblestone.
```
Target Block: 		string(4) "Lava"
New State:		string(6) "Cobblestone"
Causing Block: 		string(8) "Water"
```
- [`Lava` -> `Basalt`](https://github.com/pmmp/PocketMine-MP/blob/next-major/src/block/Lava.php#L88)
Lava (above a soul soil block) collides with blue ice and forms basalt.
```
Target Block: 		string(4) "Lava"
New State:		string(6) "Basalt"
Causing Block: 		string(8) "Blue Ice"
```
- [`Water` -> `Stone`](https://github.com/pmmp/PocketMine-MP/blob/next-major/src/block/Lava.php#L99)
A lava block flows into any water block and forces it to form into stone.
```
Target Block: 		string(4) "Water"
New State:		string(6) "Stone"
Causing Block: 		string(8) "Lava"
```